afk-daily.sh

```diff
@@ -1,1610 +1,1850 @@
+                if [ $oakRes = 0 ]; then
+                    inputTapSleep 550 1800 3
+                    oakInn_searchPresent
+                fi
+                ;;
+            1034)
+                inputTapSleep 270 1800 3
+                oakInn_searchPresent
+                if [ $oakRes = 0 ]; then
+                    inputTapSleep 550 1800 3
+                    oakInn_searchPresent
+                    if [ $oakRes = 0 ]; then
+                        inputTapSleep 690 1800 3
+                        oakInn_searchPresent
+                    fi
+                fi
+                ;;
+            1200)
+                inputTapSleep 270 1800 3
+                oakInn_searchPresent
+                if [ $oakRes = 0 ]; then
+                    inputTapSleep 410 1800 3
+                    oakInn_searchPresent
+                fi
+                ;;
+            1204)
+                inputTapSleep 270 1800 3
+                oakInn_searchPresent
+                if [ $oakRes = 0 ]; then
+                    inputTapSleep 410 1800 3
+                    oakInn_searchPresent
+                    if [ $oakRes = 0 ]; then
+                        inputTapSleep 690 1800 3
+                        oakInn_searchPresent
+                    fi
+                fi
+                ;;
+            1230)
+                inputTapSleep 270 1800 3
+                oakInn_searchPresent
+                if [ $oakRes = 0 ]; then
+                    inputTapSleep 410 1800 3
+                    oakInn_searchPresent
+                    if [ $oakRes = 0 ]; then
+                        inputTapSleep 550 1800 3
+                        oakInn_searchPresent
+                    fi
+                fi
+                ;;
+            1234)
+                inputTapSleep 270 1800 3
+                oakInn_searchPresent
+                if [ $oakRes = 0 ]; then
+                    inputTapSleep 410 1800 3
+                    oakInn_searchPresent
+                    if [ $oakRes = 0 ]; then
+                        inputTapSleep 550 1800 3
+                        oakInn_searchPresent
+                        if [ $oakRes = 0 ]; then
+                            inputTapSleep 690 1800 3
+                            oakInn_searchPresent
+                        fi
+                    fi
+                fi
+                ;;
+        esac
+    fi
+}
+
+# ##############################################################################
+# Function Name : strengthenCrystal
+# Descripton    : Strenghen Crystal
+# ##############################################################################
+strengthenCrystal() {
+    if [ $DEBUG -ge 4 ]; then echo "[DEBUG] strengthenCrystal" >&2; fi
+    if testColorOR 870 1115 f31f06; then        # If red circle
+        inputTapSleep 760 1030 3                # Resonating Crystal
+
+        # Detect if free slot, and take it.
+        testColorORTapSleep 620 1250 8ae9cf     # Detected: 8ae9cf / Not: e4c38e
```

Comment:
On today's update to the script, I actually implemented this as I had enough coins to buy another slot. Instead of testing for the blue confirm button, I decided to test for the coin icon. The reason was that I was scared the script would get confused with the possible frame of an ascended hero and mistake it for the button taking into consideration the delta value. So maybe I recommend checking instead the coin in the text area.
```suggestion
        testColorORTapSleep 380 890 efbd67     # Detected: efbd67 / Not: ?
```
Though checking this again, I realize that my check can probably also be easily confused in case the pop-up message doesn't appear... I think we need to change the way we check this still!

Reply:
I think I put my control between 2 rows/cols so it should not have any problem (or I'll need to move the control)

Reply:
I noticed you put it in between two rows. My only concern is whenever Lilith decides to change their location or change the design of it, and suddenly it isn't in between two rows anymore.

Realistically speaking though, this will most likely never happen... So I guess for now we can leave these changes as I don't know better ones currently as well. It's not ideal, but it works. :)